roles/openshift_health_checker/library/rpm_version.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+"""Ansible module for rpm-based systems determining existing package version information in a host."""
+
+from ansible.module_utils.basic import AnsibleModule
+
+IMPORT_EXCEPTION = None
+try:
+    import rpm  # pylint: disable=import-error
+except ImportError as IMPORT_EXCEPTION:
+    pass
+
+
+LIST_PKG_NAME = 0
+LIST_PKG_VERSION = 1
+
+
+class RpmVersionException(Exception):
+    """Base exception class for package version problems"""
+    def __init__(self, message, problem_pkgs=None):
+        Exception.__init__(self, message)
+        self.problem_pkgs = problem_pkgs
+
+
+def main():
+    """Entrypoint for this Ansible module"""
+    module = AnsibleModule(
+        argument_spec=dict(
+            package_list=dict(type="list", required=True),
+        ),
+        supports_check_mode=True
+    )
+
+    if IMPORT_EXCEPTION:
+        module.fail_json(msg="rpm_version module could not import rpm: %s" % IMPORT_EXCEPTION)
+
+    # determine the packages we will look for
+    pkg_list = module.params['package_list']
+    if not pkg_list:
+        module.fail_json(msg="package_list must not be empty")
+
+    # get the list of packages available and complain if any are
+    # missing or any errors occur
+    try:
+        pkg_dict = {pkg[LIST_PKG_NAME]: pkg for pkg in pkg_list}
```

Comment:
Would be great to share an example of expected usage. E.g. is it
```
rpm_version:
  package_list:
    - 0: 4.3.42-7
      1: bash
```

or 

```
rpm_version:
  package_list:
    - version: 4.3.42-7
      name: bash
```

?